crates/driver/src/extract_defrefs.rs

```diff
@@ -194,24 +196,30 @@ async fn async_extract_def_refs(
 async fn merge_defrefs(
     concurrent_io_operations: &Semaphore,
     path_sha_to_merged_defrefs: &'static Path,
-    directory: String,
+    entry: String,
     project_conf: &'static ProjectConf,
     mut work_items: Vec<ProcessedFile>,
     sha_of_conf_config: Arc<String>,
+    no_aggregate_source: bool,
```

Comment:
this worries me. In the case of proto this makes sense because protos are acyclic. But many programming languages (such as scala) are not enforced to be acyclic. Furthermore: this option can conflict with the def/ref graph (if a -> b and b -> a but no_aggregate_source == true, what do we do?).

Reply:
For now I am planning on enabling this only for Protobufs.

> (if a -> b and b -> a but no_aggregate_source == true, what do we do?).

I'd say we should fix the loop instead of the generator.